From e41d6a1d56c0e2287544d9df73b41aeeb9d864d6 Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Mon, 4 May 2020 15:29:58 +0530
Subject: [PATCH] Fix CarService crash on booting

CarService crash due to CarBleTrustAgent try to access
the SharedPreferences failed. If the user doesn't unlock
the device, the SharedPreferences is a is an encrypted file.

The CarBleTrustAgent shouldn't access the SharedPreferences
until the user unlocks the device.

Change-Id: I33a17c240768d91e93e8142a8dbde65df35519b5
Tracked-On: OAM-82632
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
---
 .../android/car/trust/CarTrustAgentEnrollmentService.java  | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git service/src/com/android/car/trust/CarTrustAgentEnrollmentService.java service/src/com/android/car/trust/CarTrustAgentEnrollmentService.java
index 7f2923da..43997b7b 100644
--- service/src/com/android/car/trust/CarTrustAgentEnrollmentService.java
+++ service/src/com/android/car/trust/CarTrustAgentEnrollmentService.java
@@ -51,6 +51,8 @@ import android.content.Context;
 import android.content.SharedPreferences;
 import android.os.IBinder;
 import android.os.RemoteException;
+import android.os.UserHandle;
+import android.os.UserManager;
 import android.util.Log;
 
 import com.android.car.BLEStreamProtos.BLEOperationProto.OperationType;
@@ -346,6 +348,11 @@ public class CarTrustAgentEnrollmentService extends ICarTrustAgentEnrollment.Stu
     @RequiresPermission(PERMISSION_CAR_ENROLL_TRUST)
     public List<TrustedDeviceInfo> getEnrolledDeviceInfosForUser(int uid) {
         ICarImpl.assertTrustAgentEnrollmentPermission(mContext);
+        if (mContext.isCredentialProtectedStorage()
+                && !mContext.getSystemService(UserManager.class)
+                .isUserUnlockingOrUnlocked(UserHandle.myUserId())) {
+            return new ArrayList<TrustedDeviceInfo>(0);
+        }
         Set<String> enrolledDeviceInfos = mTrustedDeviceService.getSharedPrefs().getStringSet(
                 String.valueOf(uid), new HashSet<>());
         List<TrustedDeviceInfo> trustedDeviceInfos = new ArrayList<>(enrolledDeviceInfos.size());
-- 
2.17.1

